File: services/src/Moe.La.Core/Entities/Litigation/LegalMemo/LegalMemoNote.cs
using System;

namespace Moe.La.Core.Entities
{
    public class LegalMemoNote : BaseEntity<int>
    {
        public int LegalMemoId { get; set; }

        public LegalMemo LegalMemo { get; set; }

        public int ReviewNumber { get; set; }

        public string Text { get; set; }

        public bool IsClosed { get; set; }

        /// <summary>
        /// The board ID.
        /// </summary>
        public int? LegalBoardId { get; set; }

        /// <summary>
        /// The board object.
        /// </summary>
        public LegalBoard LegalBoard { get; set; }

        /// <summary>
        /// The user's id who created it.
        /// </summary>
        public Guid CreatedBy { get; set; }

        /// <summary>
        /// Navigation property to the user who created it.
        /// </summary>
        public AppUser CreatedByUser { get; set; }
    }
}

File: website/Moe.La.Web/els-app/src/app/core/models/legal-memo-note.ts

export interface LegalMemoNoteList {
    id: number;
    legalMemoId: number;
    legalBoardId?: number;
    reviewNumber: number;
    text: string;
    createdOn: string;
    creationTime: string;
    createdBy: string;
    isClosed?: boolean;
}

export class LegalMemoNote {
    id?: number = 0;
    legalMemoId: number = 0;
    legalBoardId?: number = 0;
    reviewNumber?: number;
    text?: string;
    isClosed?: boolean;
}
